### PR TITLE
added RijkSearchFrame and ImageFrame abd changed some classes and tes…

### PR DESCRIPTION
**Pull request by YaffaLevy**
…t to make work

### CODE REVIEW

File: src/main/java/levy/art/json/ImageFrame.java [hunk 1]
```diff
@@ -0,0 +1,25 @@
+package levy.art.json;
+
+import javax.swing.*;
+import java.awt.*;
+import java.net.URL;
+
+public class ImageFrame extends JFrame {
+    public ImageFrame(String imageUrl) {
+        setTitle("Image Viewer");
```
Comment (andrewoid) [inline src/main/java/levy/art/json/ImageFrame.java]: The title should be the name of the art and the artist.

File: src/main/java/levy/art/json/ImageFrame.java [hunk 2]
```diff
@@ -0,0 +1,25 @@
...
+        setDefaultCloseOperation(DISPOSE_ON_CLOSE);
+        setSize(800, 600);
+
+        try {
+            URL url = new URL(imageUrl);
+            ImageIcon icon = new ImageIcon(
+                    new ImageIcon(url).getImage().getScaledInstance(800, -1, Image.SCALE_SMOOTH)
+            );
```
Comment (andrewoid) [inline src/main/java/levy/art/json/ImageFrame.java]: You're passing an ImageIcon into an ImageIcon.
Which isn't necessary.

File: src/main/java/levy/art/json/RijkSearchFrame.java [hunk 3]
```diff
@@ -0,0 +1,117 @@
...
+                        new ImageFrame(art.getImageUrl()).setVisible(true);
+                    }
+                });
+                imagesPanel.add(label);
+            } catch (Exception ex) {
+                ex.printStackTrace();
+            }
+        }
+        imagesPanel.revalidate();
+        imagesPanel.repaint();
```
Comment (andrewoid) [inline src/main/java/levy/art/json/RijkSearchFrame.java:110]: What does this do?

File: src/main/java/levy/art/json/RijkSearchFrame.java [hunk 4]
```diff
@@ -0,0 +1,117 @@
...
+                        Throwable::printStackTrace);
+    }
+
+    private void handleResponse(RijksCollection rijksCollection) {
+        imagesPanel.removeAll();
+        ArtObject[] artObjects = rijksCollection.getArtObjects();
+        for (ArtObject art : artObjects) {
+            try {
+                URL url = new URL(art.getImageUrl());
+                ImageIcon icon = new ImageIcon(new ImageIcon(url).getImage().getScaledInstance(150, 150, Image.SCALE_SMOOTH));
```
Comment (andrewoid) [inline src/main/java/levy/art/json/RijkSearchFrame.java]: This will resize it to 150x150, which will distort the photo.
You want to resize it so that it has the same aspect ratio.

File: src/main/java/levy/art/json/RijkSearchFrame.java [hunk 5]
```diff
@@ -0,0 +1,117 @@
...
+
+    private void searchArt() {
+        ApiKey apiKey = new ApiKey();
+        String keyString = apiKey.get();
+
+        if (disposable != null && !disposable.isDisposed()) {
+            disposable.dispose();
+        }
+
+        disposable = rijkService.getCollectionByQuery(keyString, searchField.getText(), currentPage)
```
Comment (andrewoid) [inline src/main/java/levy/art/json/RijkSearchFrame.java]: if searchField is empty, you need to do the other search.

File: src/main/java/levy/art/json/RijksCollection.java [hunk 6]
```diff
@@ -1,9 +1,16 @@
 package levy.art.json;
 
 import com.google.gson.annotations.SerializedName;
-import levy.art.json.ArtObject;
 
 public class RijksCollection {
-    public ArtObject[] artObjects;
-}
+    @SerializedName("artObjects")
```
Comment (andrewoid) [inline src/main/java/levy/art/json/RijksCollection.java]: not necessary

File: src/main/java/levy/art/json/RijksCollection.java [hunk 7]
```diff
@@ -1,9 +1,16 @@
...
+    private ArtObject[] artObjects;
+
+    public ArtObject[] getArtObjects() {
+        return artObjects;
+    }
 
+    public void setArtObjects(ArtObject[] artObjects) {
+        this.artObjects = artObjects;
+    }
```
Comment (andrewoid) [inline src/main/java/levy/art/json/RijksCollection.java]: This is never used.

**Comment by andrewoid** [comment]
Add screenshots and a readme

File: src/main/java/levy/art/json/ArtObject.java [hunk 8]
```diff
@@ -1,13 +1,32 @@
...
+
+    public String getImageUrl() {
+        return webImage != null ? webImage.getUrl() : null;
+    }
 
+    public WebImage getWebImage() {
+        return webImage;
+    }
+
+    public static class WebImage {
```
Comment (andrewoid) [inline src/main/java/levy/art/json/ArtObject.java]: Why did you replace ArtImage with WebImage?

File: src/main/java/levy/art/json/ImageFrame.java [hunk 9]
```diff
@@ -0,0 +1,28 @@
+package levy.art.json;
```
Comment (andrewoid) [inline src/main/java/levy/art/json/ImageFrame.java]: This doesn't belong in the json package

File: src/main/java/levy/art/json/RijkSearchFrame.java [hunk 10]
```diff
@@ -0,0 +1,133 @@
...
+                    }
+                });
+
+                imagesPanel.add(label);
+            } catch (Exception ex) {
+                ex.printStackTrace();
+            }
+        }
+        imagesPanel.revalidate();
+        imagesPanel.repaint();
```
Comment (andrewoid) [inline src/main/java/levy/art/json/RijkSearchFrame.java]: What does this do?

File: src/main/java/levy/art/json/RijksCollection.java [hunk 11]
```diff
@@ -1,9 +1,12 @@
 package levy.art.json;
 
 import com.google.gson.annotations.SerializedName;
-import levy.art.json.ArtObject;
 
 public class RijksCollection {
-    public ArtObject[] artObjects;
-}
+    private ArtObject[] artObjects;
```
Comment (andrewoid) [inline src/main/java/levy/art/json/RijksCollection.java:6]: Why did you change this from public to private?

File: src/main/java/levy/art/json/ArtObject.java (hunk 8 above)
Comment (YaffaLevy) [reply src/main/java/levy/art/json/ArtObject.java]: I just felt it better match the json name

File: src/main/java/levy/art/json/RijkSearchFrame.java (hunk 10 above)
Comment (YaffaLevy) [reply src/main/java/levy/art/json/RijkSearchFrame.java]: just a way to reformat after every search

Comment (YaffaLevy) [reply src/main/java/levy/art/json/RijkSearchFrame.java]: its not necessary i was just looking into how to organize the images and this was suggested

File: src/main/java/levy/art/json/RijksCollection.java (hunk 11 above)
Comment (YaffaLevy) [reply src/main/java/levy/art/json/RijksCollection.java:6]: didnt seem like it needed to be public

**Comment by andrewoid** [comment]
Images don't go into .github

File: src/main/java/levy/art/json/RijkSearchFrame.java [hunk 12]
```diff
@@ -0,0 +1,132 @@
+package levy.art.json;
```
Comment (andrewoid) [inline src/main/java/levy/art/json/RijkSearchFrame.java]: This also shouldn't go into .json

File: src/main/java/levy/art/json/RijkSearchFrame.java [hunk 13]
```diff
@@ -0,0 +1,132 @@
...
+                label.addMouseListener(new MouseAdapter() {
+                    @Override
+                    public void mouseClicked(MouseEvent e) {
+                        System.out.println("Yay you clicked me");
+                        new ImageFrame(art.getTitle(), art.getArtist(), art.getImageUrl()).setVisible(true);
+                    }
+                });
+
+                imagesPanel.add(label);
+            } catch (Exception ex) {
```
Comment (andrewoid) [inline src/main/java/levy/art/json/RijkSearchFrame.java]: Catch IOException, not every possible Exception

**Comment by YaffaLevy** [comment]
> Images don't go into .github

i am confused what this is reffering to

**Comment by andrewoid** [comment]
> > Images don't go into .github
> 
> i am confused what this is reffering to 

Where are you storing your screenshots?

File: src/main/java/levy/art/ImageFrame.java [hunk 14]
```diff
@@ -0,0 +1,28 @@
...
+    public ImageFrame(String title, String artist, String imageUrl) {
+        setTitle(title + " by " + artist);
+        setDefaultCloseOperation(DISPOSE_ON_CLOSE);
+        setSize(800, 600);
+
+        try {
+            URL url = new URL(imageUrl);
+            ImageIcon icon = new ImageIcon(
+                    new ImageIcon(url).getImage().getScaledInstance(800, -1, Image.SCALE_SMOOTH)
+            );
```
Comment (andrewoid) [inline src/main/java/levy/art/ImageFrame.java]: This is not how I showed you to download and resize images.

File: src/main/java/levy/art/RijkSearchFrame.java [hunk 15]
```diff
@@ -0,0 +1,134 @@
...
+        }
+    }
+
+    private void handleResponse(RijksCollection rijksCollection) {
+        imagesPanel.removeAll();
+        ArtObject[] artObjects = rijksCollection.getArtObjects();
+        for (ArtObject art : artObjects) {
+            try {
+                URL url = new URL(art.getImageUrl());
+                ImageIcon originalIcon = new ImageIcon(url);
```
Comment (andrewoid) [inline src/main/java/levy/art/RijkSearchFrame.java]: This is not how I showed you to download and resize images.

File: src/main/java/levy/art/json/ArtObject.java [hunk 16]
```diff
@@ -1,13 +1,32 @@
 package levy.art.json;
 
-import com.google.gson.annotations.SerializedName;
-import levy.art.json.ArtImage;
 
 public class ArtObject {
-    public String title;
-    public String longTitle;
-    public ArtImage webImage;
```
Comment (andrewoid) [inline src/main/java/levy/art/json/ArtObject.java:9]: Did you delete the ArtImage class?

File: src/main/java/levy/art/ImageFrame.java (hunk 14 above)
Comment (YaffaLevy) [reply src/main/java/levy/art/ImageFrame.java]: I know I had to change it around because checkstyles got upset that the line was too long

File: src/main/java/levy/art/json/ArtObject.java (hunk 16 above)
Comment (YaffaLevy) [reply src/main/java/levy/art/json/ArtObject.java:9]: no

File: src/main/java/levy/art/ImageFrame.java (hunk 14 above)
Comment (andrewoid) [reply src/main/java/levy/art/ImageFrame.java]: > I know I had to change it around because checkstyles got upset that the line was too long 

Then break up the line...

Comment (YaffaLevy) [reply src/main/java/levy/art/ImageFrame.java]: when i did it wanted me to move it back

File: src/main/java/levy/art/json/RijksCollection.java (hunk 11 above)
Comment (YaffaLevy) [reply src/main/java/levy/art/json/RijksCollection.java:6]: should it go back to public

Comment (andrewoid) [reply src/main/java/levy/art/json/RijksCollection.java:6]: > should it go back to public

It's fine this way as well.

**Review by andrewoid — COMMENTED** [review]
You added screenshots but no Readme

**Comment by YaffaLevy** [comment]
> You added screenshots but no Readme

i have readme on a sep branch called readme

**Comment by andrewoid** [comment]
> 
> > You added screenshots but no Readme
> 
> i have readme on a sep branch called readme

Why? Just put it here.

**Comment by YaffaLevy** [comment]
> > > You added screenshots but no Readme
> > 
> > 
> > i have readme on a sep branch called readme
> 
> Why? Just put it here.

because thats how you taught us to do

**Comment by andrewoid** [comment]
> because thats how you taught us to do 

Because I taught a lesson specifically on Readme.

**Comment by YaffaLevy** [comment]
> > because thats how you taught us to do
> 
> Because I taught a lesson specifically on Readme.

so I shoould switch it?

**Comment by YaffaLevy** [comment]
> > > because thats how you taught us to do
> > 
> > 
> > Because I taught a lesson specifically on Readme.
> 
> so I shoould switch it?
Can I just send a new pull request for the readme?

**Comment by andrewoid** [comment]
> Can I just send a new pull request for the readme?

We are talking about one file right? Just copy it in here.

**Comment by YaffaLevy** [comment]
> > Can I just send a new pull request for the readme?
> 
> We are talking about one file right? Just copy it in here.

yes so shoudl i deletd the readme branch or just leave and copy and paste into my other original branch

**Comment by YaffaLevy** [comment]
> > > Can I just send a new pull request for the readme?
> > 
> > 
> > We are talking about one file right? Just copy it in here.
> 
> yes so shoudl i deletd the readme branch or just leave and copy and paste into my other original branch

does that look good?

File: src/main/java/levy/art/RijkSearchFrame.java [hunk 17]
```diff
@@ -0,0 +1,131 @@
...
+        searchArt();
+    }
+
+    private void searchArt() {
+        if (disposable != null && !disposable.isDisposed()) {
+            disposable.dispose();
+        }
+
+        ApiKey apiKey = new ApiKey();
+        String keyString = apiKey.get();
```
Comment (andrewoid) [inline src/main/java/levy/art/RijkSearchFrame.java]: Do this one time. Not each time this method is called. You have a member variable for this.

Comment (andrewoid) [reply src/main/java/levy/art/RijkSearchFrame.java]: > Do this one time. Not each time this method is called. You have a member variable for this.

You are still doing this every time you search.

Comment (YaffaLevy) [reply src/main/java/levy/art/RijkSearchFrame.java]: should i only be doing it once in my service also?

Comment (YaffaLevy) [reply src/main/java/levy/art/RijkSearchFrame.java]: should i use a @beforeAll

Comment (andrewoid) [reply src/main/java/levy/art/RijkSearchFrame.java]: > should i use a @beforeAll

That's only for tests

Comment (andrewoid) [reply src/main/java/levy/art/RijkSearchFrame.java]: > should i only be doing it once in my service also? 

Once in you **application**

Comment (YaffaLevy) [reply src/main/java/levy/art/RijkSearchFrame.java]: so i shoudl set it once like in my collection and never have to use that usage again?

File: src/main/java/levy/art/RijkService.java [hunk 18]
```diff
@@ -1,11 +1,14 @@
...
 
+import com.andrewoid.ApiKey;
 import io.reactivex.rxjava3.core.Single;
 import levy.art.json.RijksCollection;
 import retrofit2.http.GET;
 import retrofit2.http.Query;
 
 public interface RijkService {
+    ApiKey apiKey = new ApiKey();
+    String keyString = apiKey.get();
```
Comment (andrewoid) [inline src/main/java/levy/art/RijkService.java]: I did not say put this in your interface.

Comment (YaffaLevy) [reply src/main/java/levy/art/RijkService.java]: so where should i put

Comment (andrewoid) [reply src/main/java/levy/art/RijkService.java]: Back where it was. But only do it once, not every time you search.

Comment (YaffaLevy) [reply src/main/java/levy/art/RijkService.java]: so what else what i need besides the beforeall?

Comment (andrewoid) [reply src/main/java/levy/art/RijkService.java]: I never said anything about beforeall

Comment (YaffaLevy) [reply src/main/java/levy/art/RijkService.java]: so how should i do it

File: src/main/java/levy/art/RijkSearchFrame.java (hunk 17 above)
Comment (YaffaLevy) [reply src/main/java/levy/art/RijkSearchFrame.java]: i put it just in my test where else should it not be?

File: src/main/java/levy/art/RijkService.java (hunk 18 above)
Comment (YaffaLevy) [reply src/main/java/levy/art/RijkService.java]: Not sure how to do that

File: src/main/java/levy/art/RijkSearchFrame.java (hunk 17 above)
Comment (andrewoid) [reply src/main/java/levy/art/RijkSearchFrame.java]: The way you had it in your test was fine. I never asked you to change that.

The way you have it in your application now is correct.

Comment (andrewoid) [inline src/main/java/levy/art/RijkSearchFrame.java]: You are already doing this on line 32, you don't need to do it here as well

Comment (YaffaLevy) [reply src/main/java/levy/art/RijkSearchFrame.java]: Screenshot 2024-05-27 at 7 04 02 PM

Comment (YaffaLevy) [reply src/main/java/levy/art/RijkSearchFrame.java]: i fixed it is this better?

Comment (YaffaLevy) [reply src/main/java/levy/art/RijkSearchFrame.java]: ?

Comment (YaffaLevy) [reply src/main/java/levy/art/RijkSearchFrame.java]: Screenshot 2024-05-28 at 10 19 04 AM
Do I need to do it all 3 times here? should I leave it? Or change?

Comment (andrewoid) [reply src/main/java/levy/art/RijkSearchFrame.java]: Yes, this has to be there all 3 times. That's how you pass the API key to the service.

**Comment by andrewoid** [comment]
Ok you put the code back the way it was, not try to fix the original thing I was telling you.

In searchArt() you create a new ApiKey. This happens every time you call that method but it's not necessary. Only do it once in your Frame (like in the constructor and save the value.

**Review by andrewoid — APPROVED** [review]
